src/tbp/monty/frameworks/models/evidence_matching/resampling_hypotheses_updater.py

```diff
@@ -384,6 +386,8 @@ def _sample_count(
                 evidence, locations, and poses based on the input channel
             tracker: Slope tracker for the evidence values of a
                 graph_id
+            init_hyp_space: Initialize a new hypothesis space. Happens only at the
```

Comment:
Wouldn't this also happen if a hypothesis space went down to zero, got deleted, and then hypotheses for the object are resampled again?